src/DeviceDiscoveryExtension/DDDevice.cs

```diff
@@ -0,0 +1,31 @@
+//
+// Custom methods for DDDevice
+//
+// Authors:
+//   Israel Soto (issoto@microsoft.com)
+//
+// Copyright 2022 Microsoft Corporation.
+//
+
+#nullable enable
+
+using ObjCRuntime;
+using Foundation;
+using Network;
+using System;
+
+namespace DeviceDiscoveryExtension {
+	public partial class DDDevice {
+
+		NWEndpoint? networkEndpoint = null;
```

Comment:
Why do you need to store the property in a field? It seems it's easier (and more correct) to create a new NWEndpoint instance in the getter (and just assign to the internal _NetworkEndpoint property in the setter).

Potential problems with current implementation:

* If the property in native code changes without going through managed code, we'll still return the field value.
* Assigning the same instance fails:

```cs
device.NetworkEndpoint = new NWEndpoint (...);
device.NetworkEndpoint = device.NetworkEndpoint;
```

Since the assigned value and the old value is the same, the setter would dispose the assigned value, and end up sending IntPtr.Zero to the native property.